src/main/kotlin/com/exactpro/th2/common/message/MessageBuilder.kt

```diff
@@ -0,0 +1,105 @@
+/*
+ * Copyright 2021-2021 Exactpro (Exactpro Systems Limited)
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.exactpro.th2.common.message
+
+import com.exactpro.th2.common.grpc.ConnectionID
+import com.exactpro.th2.common.grpc.EventID
+import com.exactpro.th2.common.grpc.MessageID
+import com.google.protobuf.Timestamp
+import java.time.Instant
+
+abstract class MessageBuilder<T> {
```

Comment:
Please try to implement type-safe-builders
https://kotlinlang.org/docs/type-safe-builders.html#full-definition-of-the-com-example-html-package